cypress/components/menu/menu.cy.js

```diff
@@ -1569,8 +1569,10 @@ context("Testing Menu component", () => {
         .eq(positionOfElement("first"), div)
         .trigger("mouseover")
         .then(() => {
-          // eslint-disable-next-line no-unused-expressions
-          expect(callback).to.have.been.calledOnce;
+          cy.wait(0).then(() => {
```

Comment:
why do we need to wait for (0) seconds? :) for all tests below?

Reply:
hah you're quick Igor! I expected to have this conversation later when it reached QA 😅 

Basically it's because the fix involved adding a `useEffect` which means the submenu only opens on the next render (which happens straight away), rather than straight away in the event handler which it was before. For that reason, without the wait these tests all fail - the callback is called but a tiny fraction of a second after the assertion. Thankfully a wait of 0 does work, it just needs to be told to run asynchronously.

I don't particularly like the way I was forced to write this test though, particularly the nested `then` (just putting `cy.wait(...)` above didn't work) - so I'd be happy to hear better ways of writing it!

Reply:
I've taken a look at the cypress docs and I noticed they generally [don't recommend chaining off of actions](https://docs.cypress.io/guides/core-concepts/retry-ability#Incorrectly-checking-if-the-stub-was-called) like `trigger()`.

Using [an alias for the callback](https://docs.cypress.io/guides/core-concepts/retry-ability#Correctly-waiting-for-the-stub-to-be-called) should work instead to avoid having to use `cy.wait`. This should take into account React re-rendering the DOM because of the effect:

```tsx
// in describe block...
const callback = cy.stub().as("callback");

// in test...
submenu().eq(positionOfElement("first"), div).trigger("mouseover");
cy.get("@callback").should("have.been.calledOnce");
```